libs/frontend/feature-render-builder/src/components/ProseMirrorEditor.ts

```diff
@@ -201,12 +205,44 @@ export const ProseMirrorEditor = defineComponent({
       },
     )
 
+    const cursorLinkAttrs = ref()
+
+    watch(editViewTxCount, () => {
+      const { editView } = editor.value
+      if (editView) {
+        const linkMark = firstMarkInSelection(editView.state, schemaText.marks.link)
+        cursorLinkAttrs.value = linkMark?.attrs
+      }
+    })
+
     return () => {
-      return h('div', {
-        id: containerId.value,
-        ref: container,
-        class: 'prose-mirror-editor-container',
-      })
+      return h(
+        'div',
+        {
+          id: containerId.value,
+          ref: container,
+          class: 'prose-mirror-editor-container',
+        },
+        cursorLinkAttrs.value
+          ? h(
+              // Use teleport to prevent LinkTooltip from constantly changing position
+              // due to the relative/absolute layout in the builder.
+              Teleport,
```

Comment:
Use teleport here to prevent `LinkTooltip` from flying around due to the absolute/relative layout in the builder (i.e. `force-relative` class).

https://github.com/samatechtw/pubstudio-builder/assets/59676941/16703b58-3824-43e2-bd09-680c77550149

Reply:
Sounds good! I moved the Teleport to `LinkComponent`, because I think it's fine to include in both situations (prosemirror and builder links).